client/containers/route/action-creator.js

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2021 Uber Technologies Inc.
+//
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import { ROUTE_PUSH, ROUTE_REPLACE, ROUTE_UPDATE_QUERY } from './action-types';
+import { ROUTE_QUERY } from './getter-types';
+
+const actionCreator = router => ({
+  [ROUTE_PUSH]: (_, args) => router.push(args),
+  [ROUTE_REPLACE]: (_, args) => router.replace(args),
+  [ROUTE_UPDATE_QUERY]: ({ getters }, args) => {
```

Comment:
I presume we have just three actions to change the current route

Reply:
Yes thats right, they are similar but have slightly different effect for a user:

**ROUTE_PUSH** will push a new route onto the router stack. This effectively will route users to the new route. This is useful when you **do** need a user to navigate to the previous screen via browser back button.

**ROUTE_REPLACE** will replace the existing route with the new route or query params etc. This effectively will route users to the new route. This is useful when you **don't** want the browser back button to navigate to the previous route.

**ROUTE_UPDATE_QUERY** is similar to ROUTE_REPLACE, except it takes the current routes' queries and only overrides what has been specified. It will have the same effect if a user tries to use the browser back button.

You can read more about these actions here:
https://router.vuejs.org/api/#router-push

Let me know if anything is unclear.
Thanks!